esmvalcore/cmor/check.py

```diff
@@ -436,13 +436,33 @@ def _check_time_coord(self):
             self.report_error(self._does_msg, var_name,
                               'have time reference units')
         else:
+            old_units = coord.units
             coord.convert_units(
                 cf_units.Unit(
-                    'days since 1950-1-1 00:00:00',
+                    'days since 1850-1-1 00:00:00',
                     calendar=coord.units.calendar))
             simplified_cal = self._simplify_calendar(coord.units.calendar)
             coord.units = cf_units.Unit(coord.units.origin, simplified_cal)
 
+            attrs = self._cube.attributes
+            branch_child = 'branch_time_in_child'
+            if branch_child in attrs:
+                attrs[branch_child] = old_units.convert(attrs[branch_child],
+                                                        coord.units)
+
+            parent_time = 'parent_time_units'
+            if parent_time in attrs:
+                parent_units = cf_units.Unit(attrs[parent_time],
+                                             simplified_cal)
+                attrs[parent_time] = cf_units.Unit(
+                    'days since 1850-1-1 00:00:00',
+                    calendar=coord.units.calendar)
```

Comment:
Did this work for you? I get
```
TypeError: illegal data type for attribute b'parent_time_units', must be one of dict_keys(['S1', 'i1', 'u1', 'i2', 'u2', 'i4', 'u4', 'i8', 'u8', 'f4', 'f8']), got O
```
from the netCDF4 library. I think the attribute needs to be only the string, ie
```
attrs[parent_time] = 'days since 1850-1-1 00:00:00'
```

Reply:
That's true, sorry I was just checking if the tests were running fine instead of using actual data.

Reply:
Looking good now.